### PR TITLE
Retain the StorageClass of the file when copying it.

### PR DESCRIPTION
**Pull request by andris-sevcenko**

### CODE REVIEW

File: src/AwsS3Adapter.php [hunk 1]
```diff
@@ -409,6 +411,20 @@ public function copy($path, $newpath)
     }
 
     /**
+     * Return an object's storage class.
+     *
+     * @param string $path
+     *
+     * @return string
+     */
+    public function getStorageClass($path)
```
Comment (frankdejonge) [inline src/AwsS3Adapter.php]: This should be private or protected.

**Comment by andris-sevcenko** [comment]
Any updates on this?

**Comment by andris-sevcenko** [comment]
So uhh, hate to be a jerk about this, but if this is all cool can this be merged?

**Comment by andris-sevcenko** [comment]
@frankdejonge so uh. Anything?

**Comment by frankdejonge** [comment]
Hi, sorry for not attending to this sooner. But have you tried getting this functionality covered in the SDk itself? As in, opening an issue there. This really seems like something they should handle. Handling this in this adapter is a quite substantial performance killer.

**Comment by andris-sevcenko** [comment]
Huh. Good call. This is super inefficient. I'll look into other options.

**Pull request closed without merging**